Core/src/Testing/DatastoreOperationRefreshTrait.php

```diff
@@ -57,13 +60,87 @@ public function refreshOperation($stub, ConnectionInterface $connection, array $
             $options['returnInt64AsObject']
         );
 
+        $serializer = $this->getSerializer();
+
         $stub->___setProperty('operation', new Operation(
-            $connection,
+            $requestHandler,
+            $serializer,
             $options['projectId'],
             $options['returnInt64AsObject'],
             $mapper
         ));
 
         return $stub;
     }
+
+    /**
+     * Helper method for Unit and Snippet test classes. This mocks the
+     * $requestHandler class property present in the Test Class with
+     * given arguments.
+     *
+     * @param string $methodName The method name to mock in RequestHandler::sendRequest
+     * @param array<string, mixed> $params The parameters to look for in the
+     * array equivalent of rpc request.
+     * @param mixed $returnValue The value to be returned by sendRequest mock.
+     * @param null|int $shouldBeCalledTimes Adds a shouldBeCalled prophecy. Defaults to `null`, implying nothing is added.
+     * [
+     *      `0` => `shouldBeCalled`,
+     *      Non zero positive integer $x => `shouldBeCalledTimes($x)`
+     * ]
+     */
+    private function mockSendRequest($methodName, $params, $returnValue, $shouldBeCalledTimes = null)
```

Comment:
Makes sense. 

There were a total of 101 tests which used sendRequest mocking. I saw that there are at least 3 lines which are repeated for every mocking (the `DatastoreClient::class`, `apiName`, and the one where we converted the request to it's array counterpart.). So I went with reducing the repeating code by 300 lines.

But yeah, if we see the library from a perspective where a developer just wants to add/update a test, then it would present a bigger onboarding and maintenance step compared to the way where we have mocks in the same test class itself.

I've updated all the tests utilizing the `mockSendRequest` method via https://github.com/yash30201/google-cloud-php/pull/9